EU4toV2/Source/Mappers/CultureGroupsMapper/CultureGroupsMapper.cpp

```diff
@@ -0,0 +1,39 @@
+#include "CultureGroupsMapper.h"
+#include "CommonRegexes.h"
+#include "CultureGroupsMapping.h"
+#include "Log.h"
+#include "ParserHelpers.h"
+
+mappers::CultureGroupsMapper::CultureGroupsMapper()
+{
+	Log(LogLevel::Info) << "-> Loading culture group equivalences.";
+	registerKeys();
+	parseFile("configurables/culture_group_map.txt");
+	clearRegisteredKeywords();
+	Log(LogLevel::Info) << "-> " << eu4ToV2CultureGroups.size() << " links loaded.";
+}
+
+mappers::CultureGroupsMapper::CultureGroupsMapper(std::istream& theStream)
+{
+	registerKeys();
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void mappers::CultureGroupsMapper::registerKeys()
+{
+	registerKeyword("link", [this](std::istream& theStream) {
+		const CultureGroupsMapping mapping(theStream);
+		if (!mapping.getV2().empty() && !mapping.getEU4s().empty())
```

Comment:
Checking for eu4 mappings is redundant here. The loop below will do nothing if its empty.